src/legacy/shrink.rs

```diff
@@ -0,0 +1,426 @@
+use std::collections::VecDeque;
+use std::io::{self, copy, Read};
+
+use thiserror::Error;
+
+use super::bitstream::{lsb, BitStream};
+
+const MIN_CODE_SIZE: u8 = 9;
+const MAX_CODE_SIZE: u8 = 13;
+
+const MAX_CODE: usize = (1 << MAX_CODE_SIZE) - 1;
+const INVALID_CODE: u16 = u16::MAX;
+const CONTROL_CODE: usize = 256;
+const INC_CODE_SIZE: u64 = 1;
+const PARTIAL_CLEAR: u64 = 2;
+
+// const HASH_BITS: usize = MAX_CODE_SIZE + 1; /* For a load factor of 0.5. */
+// const HASHTAB_SIZE: usize = 1 << HASH_BITS;
+const UNKNOWN_LEN: u16 = u16::MAX;
+
+#[derive(Error, Debug)]
+enum ShrinkError {
```

Comment:
Again, `ZipError::InvalidArchive(&'static str)` will probably suffice.

Reply:
moved to io::result